openadapt/strategies/stateful.py

```diff
@@ -143,19 +145,26 @@ def get_window_state_diffs(
     ignore_window_ids = set()
     if ignore_boundary_windows:
         first_window_event = action_events[0].window_event
-        first_window_id = first_window_event.state["window_id"]
-        first_window_title = first_window_event.title
+        if first_window_event.state :
+            first_window_id = first_window_event.state["window_id"]
+        else :
+            first_window_id = None
+            first_window_title = first_window_event.title
         last_window_event = action_events[-1].window_event
-        last_window_id = last_window_event.state["window_id"]
-        last_window_title = last_window_event.title
+        if last_window_event.state :
+            last_window_id = last_window_event.state["window_id"]
+            last_window_title = last_window_event.title
+        else :
+            last_window_id = None
+            last_window_title = last_window_event.title
         if first_window_id != last_window_id:
             logger.warning(f"{first_window_id=} != {last_window_id=}")
         ignore_window_ids.add(first_window_id)
         ignore_window_ids.add(last_window_id)
         logger.info(f"ignoring {first_window_title=} {last_window_title=}")
     window_event_states = [
         action_event.window_event.state
-        if action_event.window_event.state["window_id"] not in ignore_window_ids
+        if action_event.window_event.state is not None and action_event.window_event.state["window_id"] not in ignore_window_ids
```

Comment:
I think we can remove `is not None` here

Reply:
addressed in https://github.com/OpenAdaptAI/OpenAdapt/pull/290/commits/1027546e62c40ef546e8c8a36050d46d8ff79c97